app/src/main/java/co/stonephone/stonecamera/plugins/VideoMode.kt

```diff
@@ -0,0 +1,185 @@
+package co.stonephone.stonecamera.plugins
+
+import android.util.Log
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.FlipCameraAndroid
+import androidx.compose.material.icons.filled.Pause
+import androidx.compose.material.icons.filled.PlayArrow
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.RectangleShape
+import androidx.compose.ui.unit.dp
+import co.stonephone.stonecamera.StoneCameraViewModel
+
+class VideoModePlugin : IPlugin {
+    override val id: String = "videoMode"
+    override val name: String = "Video Mode"
+
+    private lateinit var viewModel: StoneCameraViewModel
+
+    override fun onModeSelected(
+        viewModel: StoneCameraViewModel,
+        previousMode: String,
+        nextMode: String
+    ) {
+        if (previousMode == modeLabel) {
+            viewModel.stopRecording()
+        }
+    }
+
+    override val modeUseCases: List<PluginUseCase>
+        get() = listOf(PluginUseCase.VIDEO, PluginUseCase.PHOTO, PluginUseCase.ANALYSIS)
+
+    override val modeLabel
+        get() = "video"
+
+    override val renderModeControl
+        get() = @Composable {
+
+            val isRecording by viewModel::isRecording
+            val isPaused by viewModel::isPaused
+
+            // Bottom row (Flip + Shutter)
+            Row(
+                modifier = Modifier.fillMaxWidth(),
+                horizontalArrangement = Arrangement.SpaceEvenly,
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                // Camera Switcher Button
```

Comment:
I'm very glad these comments exist, otherwise it'd be vary difficult to guess what the code does.